frontend/packages/kubevirt-plugin/src/components/create-vm-wizard/redux/validations/vm-settings-tab-validation.ts

```diff
@@ -108,6 +115,27 @@ const asVMSettingsFieldValidator = (
     subject: getFieldTitle(iGetFieldKey(field)),
   });
 
+const validateSource: Validator = (field, options): ValidationObject => {
+  const value = iGetFieldValue(field);
+  const emptyError = validateEmptyValue(value);
+  if (emptyError) {
+    return emptyError;
+  }
+
+  if (value === ProvisionSource.PXE.getValue()) {
+    const { getState, id } = options;
+    const state = getState();
+    const nads = iGetCommonData(state, id, VMWizardProps.nadsData);
+    const hasNADs =
+      !iGetLoadError(nads) && immutableListToShallowJS(iGetLoadedData(nads))?.length !== 0;
```

Comment:
you can find the length without converting the whole array to JS

Reply:
does this take into account loading?

Reply:
> does this take into account loading?

yes, `iGetLoadedData` checks for `loading` internally so we can't get a false positive while loading.

Reply:
this will hapen when you use a template with PXE and nads have not loaded yet

![aa](https://user-images.githubusercontent.com/3648838/93898379-734e8680-fcf3-11ea-97d5-42bb453e1de9.png)

Reply:
are you saying we should not validate fields field in by templates ?